src/plugins/initiatives/discourse.test.js

```diff
@@ -0,0 +1,271 @@
+// @flow
+
+import {groupURLsByHeader, initiativeFromDiscourseTracker} from "./discourse";
+
+describe("plugins/initiatives/discourse", () => {
+  describe("initiativeFromDiscourseTracker", () => {
+    it("handles an example tracker", async () => {
+      // Given
+      const serverUrl = "https://foo.bar";
+      const topic = {
+        id: 123,
+        title: "Example initiative",
+        timestampMs: 1571498171951,
+        authorUsername: "TestUser",
+      };
+      const firstPost = {
+        id: 432,
+        topicId: 123,
+        indexWithinTopic: 1,
+        replyToPostIndex: null,
+        timestampMs: 1571498171951,
+        authorUsername: "TestUser",
+        cooked: `
+            <h1>Example initiative</h1>
+            <h2>Status: Testing</h2>
+            <h2>Champion<a href="https://foo.bar/t/dont-include/10"><sup>?</sup></a>:</h2>
+            <p>
+              <a class="mention" href="/u/ChampUser">@ChampUser</a>
+            </p>
+            <h2>Dependencies:</h2>
+            <ul>
+              <li><a href="https://foo.bar/t/dependency/1">Thing we need</a></li>
+              <li><a href="https://foo.bar/t/dependency/2">Thing we need</a></li>
+              <li><a href="https://foo.bar/t/dependency/3">Thing we need</a></li>
+            </ul>
+            <h2>References:</h2>
+            <ul>
+              <li><a href="https://foo.bar/t/reference/4">Some reference</a></li>
+              <li><a href="https://foo.bar/t/reference/5/2">Some reference</a></li>
+              <li><a href="https://foo.bar/t/reference/6/4">Some reference</a></li>
+            </ul>
+            <h2>Contributions:</h2>
+            <ul>
+              <li><a href="https://foo.bar/t/contribution/7">Some contribution</a></li>
+              <li><a href="https://foo.bar/t/contribution/8/2">Some contribution</a></li>
+              <li><a href="https://github.com/sourcecred/sourcecred/pull/1416">Some contribution</a></li>
+            </ul>
+          `,
+      };
+
+      // When
+      const initiative = await initiativeFromDiscourseTracker(
+        serverUrl,
+        topic,
+        firstPost
+      );
+
+      // Then
+      expect(initiative.completed).toEqual(false);
+      expect(initiative).toMatchSnapshot();
```

Comment:
IMO, this would be a good fit for `toMatchInlineSnapshot` -- makes it much easier to check the output against the test case.

Reply:
also, note that right now the snapshot is interpreted as a binary file, which makes it quite hard to code review. this is likely because of the raw `NodeAddressT` in the initiative, which contains null bytes.

solutions that might work:
- `JSON.stringify` the initiative
- use NodeAddress.toString on the node address before snapshotting

Reply:
The binary interpretation was why I didn't inline. I'll change it to the JSON version. :+1:

Reply:
The JSON version added a lot of escaping cruft, went for a `NodeAddress.toParts` approach.